task/buildpacks-phases/0.3/buildpacks-phases.yaml

```diff
@@ -0,0 +1,486 @@
+      description: Reference to a cache image in an OCI registry (if no cache workspace is provided).
+      default: ""
+    - name: CNB_ENV_VARS
+      type: array
+      description: Environment variables to set during _build-time_.
+      default: []
+    - name: CNB_EXPERIMENTAL_MODE
+      description: Control the lifecycle's execution according to the mode silent, warn, error for the experimental features.
+      default: silent
+    - name: CNB_GROUP_ID
+      description: The group ID of the builder image user.
+      default: ""
+    - name: CNB_INSECURE_REGISTRIES
+      description: List of registries separated by a comma having a self-signed certificate where TLS verification will be skipped.
+      default: ""
+    - name: CNB_LAYERS_DIR
+      description: Path to layers directory
+      default: "/layers"
+    - name: CNB_LOG_LEVEL
+      description: Logging level values info, warning, error, debug
+      default: "info"
+    - name: CNB_PLATFORM_API_SUPPORTED
+      description: Buildpack Platform API supported by the Tekton task
+      default: "0.13"
+    - name: CNB_PLATFORM_API
+      description: User's Buildpack Platform API
+      default: ""
+    - name: CNB_PLATFORM_DIR
+      description: Path to the platform directory
+      default: "/platform"
+    - name: CNB_PROCESS_TYPE
+      description: Default process type to set in the exported image
+      default: "web"
+    - name: CNB_RUN_IMAGE
+      description: Reference to an image which is packaging the application runtime to be launched.
+      default: ""
+    - name: CNB_SKIP_LAYERS
+      description: Do not restore SBOM layer from previous image
+      default: false
+    # DEPRECATED: It does not make sense to support such an env variable as mounting the unix docker socket part of a pod from a host volume
+    # will never happen for security reason
+    # - name: CNB_USE_DAEMON
+    #  description: Analyze image from docker daemon
+    #  default: false
+    - name: CNB_USER_ID
+      description: The user ID of the builder image user.
+      default: ""
+
+    - name: APP_IMAGE
+      description: The name of the container image for your application.
+    - name: SOURCE_SUBPATH
+      description: A subpath within the `source` input where the source to build is located.
+      default: ""
+    - name: TAGS
+      description: Additional tag to apply to the exported image
+      default: ""
+    - name: USER_HOME
+      description: Absolute path to the user's home directory.
+      default: /tekton/home
+
+  results:
+    - name: APP_IMAGE_DIGEST
+      description: The digest of the built `APP_IMAGE`.
+
+  stepTemplate:
+    env:
+      - name: CNB_EXPERIMENTAL_MODE
+        value: $(params.CNB_EXPERIMENTAL_MODE)
+      - name: HOME
+        value: $(params.USER_HOME)
+
+  steps:
+    - name: get-labels-and-env
+      image: quay.io/halkyonio/skopeo-jq:0.1.3@sha256:1b3d21ad541227dc9d3e793d18cef9eb00a969c0c01eb09cab88997bc63680c6
```

Comment:
as we are giving others the option to build their own image and use, should we parameterize this image?

Reply:
This is an image packaging skopeo + jq that we need here to inspect the content of the image. We cannot replace it

Reply:
@cmoulliard, agreed but what if current image get vulnerabilities? User of this task should able to use different image without actually modifying the task, I believe.

Reply:
So I will set it as default to a new parameter that I will create. Nevertheless, If vulnerabilities are discovered by the way, then a new task should be created in this case ...

Reply:
>So I will set it as default to a new parameter that I will create.

yeah that's the ideal way.

>If vulnerabilities are discovered by the way, then a new task should be created in this case.

Ideally new version of the task is required if there are any breaking changes that we want to push into the task and might break existing users.

>The konflux project uses a tool able to discover which requires to be changed and creates PR. This project could maybe uses it in order to guarantee that all the referenced images are valid, etc

That's a great idea but I wonder if as tekton team we would be able to maintain that for a long run (especially for this project as it's very less maintained 😅)
I would let @vdemeester and @afrittoli chime in and help take that decision 😅

Reply:
> That's a great idea but I wonder if as tekton team we would be able to maintain that f

This is proposition that I'm doing for the whole catalog ;-)